presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/DetermineJoinDistributionType.java

```diff
@@ -224,6 +230,13 @@ private boolean isSmall(PlanNode node)
             return !dataSize.isValueUnknown() && dataSize.getValue() < smallSizeLimit;
         }
 
+        private boolean isMuchSmaller(PlanNode node, PlanNode other)
+        {
+            Estimate nodeDataSize = getOutputSizeEstimate(node);
+            Estimate otherDataSize = getOutputSizeEstimate(other);
+            return !nodeDataSize.isValueUnknown() && !otherDataSize.isValueUnknown() && nodeDataSize.getValue() * (nodeCount - 1) < otherDataSize.getValue();
```

Comment:
why is it `nodeCount - 1`? It should be:

`BUILD < PROBE / n` <=> `nodeDataSize.getValue() * nodeCount < otherDataSize.getValue()`